### PR TITLE
Fix setup of argument variables in case parameter names are duplicated

### PR DESCRIPTION
**Pull request by ruben-ayrapetyan**

### CODE REVIEW

File: tests/jerry/function-args.js [hunk 1]
```diff
@@ -44,3 +44,9 @@ for(i = 11; i <= 20; i++)
 {
   f1(i);
 }
+
+function g (p, p) {
+  assert (p === 'second');
+}
+
```
Comment (galpeter) [inline tests/jerry/function-args.js]: Maybe we should have a test also with the arguments object also, like:
```js
assert (p  === arguments[1]);
```

Comment (ruben-ayrapetyan) [inline tests/jerry/function-args.js]: The test is added.

**Comment by ruben-ayrapetyan** [comment]
Pull request is rebased.

**Comment by galpeter** [comment]
looks good to me

**Comment by zherczeg** [comment]
lgtm

**Comment by zherczeg** [comment]
Btw the engine suffers a lot from these reversed order construction mechanics. Could we globally change that?

**Comment by ruben-ayrapetyan** [comment]
> Btw the engine suffers a lot from these reversed order construction mechanics. Could we globally change that?

@zherczeg, could you, please, provide more details on the issue?

**Comment by egavrin** [comment]
@zherczeg @ruben-ayrapetyan folks, I just raised discussion for that #287.

**Comment by egavrin** [comment]
`make push`

**Comment by ruben-ayrapetyan** [comment]
> @zherczeg @ruben-ayrapetyan folks, I just raised discussion for that #287.

@egavrin, great! Thank you!

**Pull request merged**